src/Fixer/Comment/SingleLineCommentStyleFixer.php

```diff
@@ -0,0 +1,123 @@
+        $this->hashEnabled = in_array('hash', $this->configuration['comment_types'], true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Single-line comments and multi-line comments with only one line of actual content should use the `//` syntax.',
+            [
+                new CodeSample(
+                    "<?php\n/* first comment */\n\$a = 1;\n/*\n * second comment\n */\n\$b = 2;\n/*\n * third\n * comment\n */\n\$c = 3;",
+                    ['comment_types' => ['star']]
+                ),
+                new CodeSample(
+                    '<?php # comment',
+                    ['comment_types' => ['hash']]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_COMMENT)) {
+                continue;
+            }
+            $content = $token->getContent();
+            $commentContent = substr($content, 2, -2);
+            if ($this->hashEnabled && '#' === $content[0]) {
+                $tokens[$index] = new Token([$token->getId(), '//'.substr($content, 1)]);
+                continue;
+            }
+            if (!$this->starEnabled || '/*' !== substr($content, 0, 2) || preg_match('/[^\s\*].*\R.*[^\s\*]/s', $commentContent)) {
+                continue;
+            }
+            $nextTokenIndex = $index + 1;
+            if (isset($tokens[$nextTokenIndex])) {
+                $nextToken = $tokens[$nextTokenIndex];
+                if (!$nextToken->isWhitespace() || !preg_match('/\R/', $nextToken->getContent())) {
+                    continue;
+                }
+
+                $tokens[$nextTokenIndex] = new Token([$nextToken->getId(), ltrim($nextToken->getContent(), " \t")]);
+            }
+
+            $content = '//';
+            if (preg_match('/[^\s\*]/', $commentContent)) {
+                $content = '// '.preg_replace('/[\s\*]*([^\s\*](.+[^\s\*])?)[\s\*]*/', '\1', $commentContent);
+            }
+            $tokens[$index] = new Token([$token->getId(), $content]);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('comment_types', 'List of comment types to fix'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([
+                    (new FixerOptionValidatorGenerator())->allowedValueIsSubsetOf(['star', 'hash']),
+                ])
+                ->setDefault(['star', 'hash'])
```

Comment:
I think `star` is ok, and it's what I think of when I see a `*. Kleene star, etc. :P